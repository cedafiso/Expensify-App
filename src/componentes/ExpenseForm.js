import React from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker} from 'react-dates';
import { addExpense } from '../actions/expenses';
import { connect } from 'react-redux';


export default class ExpenseForm extends React.Component {
    state = {
        description: this.props.expense ? this.props.expense.description : '',
        amount: this.props.expense  ? (this.props.expense.amount/100).toString() : '',
        note: this.props.expense  ? this.props.expense.note : '',
        createdAt: this.props.expense  ? moment(this.props.expense.createdAt) : moment(),
        focused: false,
        error:'',
    };
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({description}));
    }
    onAmountChange = (e) => {
        const amount = e.target.value;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(() => ({amount}));
        }else{
            return null;
        }
        
    }
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({note}));
    }
    onDateChange = (createdAt) => {
        if(createdAt){
            this.setState(() => ({createdAt}));
        }
    }
    onFocusChange = ({focused}) => {
        this.setState(() => ({focused}));
    }
    onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.amount || !this.state.description){
            this.setState(() => ({error: 'Please provide description and amount'}));
        }else{
           this.setState(() => {error:''});
           this.props.onSubmit({
               description: this.state.description,
               amount: parseFloat(this.state.amount, 10)*100 ,
               createdAt: this.state.createdAt.valueOf(),
               note: this.state.note
           });
          }
    };
    render(){
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    {this.state.error && <p>{this.state.error}</p>}
                    <input 
                    type='text'
                    placerholder='Description'
                    value={this.state.description}
                    onChange={this.onDescriptionChange}
                    autoFocus/>
                    <input 
                    type='text'
                    placeholder='Amount'
                    value={this.state.amount}
                    onChange={this.onAmountChange}

                    />
                    <SingleDatePicker
                    date ={this.state.createdAt}
                    onDateChange={this.onDateChange}
                    focused={this.state.focused}
                    onFocusChange={this.onFocusChange}
                    />
                    <textarea 
                    placeholder="Add a note for your expense (optional)"
                    value={this.state.note}
                    onChange={this.onNoteChange}
                    />
                    <button>Submit</button>
                </form>
            </div>
        )
    }

}

connect()(ExpenseForm);