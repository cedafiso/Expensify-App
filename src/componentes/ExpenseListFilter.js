import React from 'react';
import {connect} from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import {DateRangePicker} from 'react-dates';

class ExpenseListFilter extends React.Component{
    state = {
        focusedInput: null,
    }
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }
    onFocusChange = (focusedInput) => {
        this.setState({focusedInput})
    }
    render(){
        return (
            <div>
                <input 
                type='text' 
                value={this.props.filters.text} 
                onChange={(e)=>{
                    return this.props.dispatch(setTextFilter(e.target.value))
                    }}>
    
                </input>
                <p>Sort by:</p>
                <select value={this.props.filters.sortBy} 
                        onChange={(e) => {
                            if (e.target.value === 'amount'){
                                return this.props.dispatch(sortByAmount());
                            }else if(e.target.value === 'date'){
                                return this.props.dispatch(sortByDate());
                            }
                        }}>
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
                <DateRangePicker
                startDate={this.props.filters.startDate}
                startDateId='Start_Date'
                endDate={this.props.filters.endDate}
                endDateId='End_Date'
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.focusedInput}
                onFocusChange={this.onFocusChange}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={() => false}
                />
            </div>
        )
    }

}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

export default connect(mapStateToProps)(ExpenseListFilter);