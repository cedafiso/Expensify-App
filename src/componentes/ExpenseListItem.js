import React from 'react';
import {connect} from 'react-redux';
import { startRemoveExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({description , amount, createdAt, id, dispatch}) => {
    return(
        <div>
            <p>Description: {description}</p>
            <p>Amount: {numeral(amount/100).format('$0,0.00')}</p>
            <p>Created at: {moment(createdAt).format("MMMM Do, YYYY")}</p>
            <button onClick={(e) => {
                return dispatch(startRemoveExpense({id}))}}>
                Remove
            </button>
            <Link to={'/edit/'+id}>
                <button>
                Edit
                </button>
            </Link>
        </div>
    )
};



export default connect()(ExpenseListItem);