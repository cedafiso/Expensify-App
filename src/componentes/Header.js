import React from 'react';
import {NavLink, Link} from 'react-router-dom';
import { startLogOut } from '../actions/auth';
import {connect} from 'react-redux';
const Header = ({startLogOut}) => {
    return (
        <header className="header">
            <div className="content-container">
                <div className="header__content">
                    <Link className="header__tittle" to="/dashboard">
                        <h1>Expensify-App</h1>
                    </Link>
                    <button className="button button--logout"onClick={startLogOut}>Log out</button>
                </div>
            </div>
        </header>
    );
}

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
});
export default connect(undefined, mapDispatchToProps)(Header);