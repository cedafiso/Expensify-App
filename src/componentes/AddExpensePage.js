import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startAddExpense} from '../actions/expenses';

const AddExpensePage = (props) => {
    return (
        <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__tittle">Add expense</h1>
                </div>
            </div>
            <div className="content-container">
                <ExpenseForm 
                onSubmit={(expenses) => {
                    props.dispatch(startAddExpense(expenses));
                    props.history.push('/');
                }}
                />
            </div>
        </div>
    );
}

export default connect()(AddExpensePage);