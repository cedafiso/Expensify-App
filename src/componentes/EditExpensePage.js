import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {editExpense} from '../actions/expenses';
import {useParams} from 'react-router-dom';

const EditExpensePage = (props) => {
    console.log(props.expenses);
    return (
        <div>
            <p>Expensify-App-Edit {props.match.params.id}</p>
            <ExpenseForm
                expense={props.expenses}
                onSubmit={(expenses) => {
                props.dispatch(editExpense(props.match.params.id, expenses));
                props.history.push('/');
            }
                }/>
        </div>
    );
}

const mapStateToProps = (state, props) => ({
    expenses: state.expenses.find((expense) => {
        return expense.id === props.match.params.id})
});

export default connect(mapStateToProps)(EditExpensePage);