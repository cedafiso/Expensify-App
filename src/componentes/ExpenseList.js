import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import visibleExpenses from '../selectors/expenses';


const ExpenseList = (props) => {
    return (
        <div>
            <h1>Expense List:</h1>
            {props.expenses.map((item) => {
                return (<ExpenseListItem
                    key={item.id}
                    {...item}

                />)
            })}
        </div>
    )
};

const ConnectedExpenseList = connect()

const mapStateToProps = (state) => {
    return {
        expenses : visibleExpenses(state.expenses , state.filters)
    };
}

export default connect(mapStateToProps)(ExpenseList);