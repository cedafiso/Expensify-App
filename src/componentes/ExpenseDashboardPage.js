import React from 'react';
import ExpenseList from './ExpenseList';

const ExpenseDashboardPage = () => {
    return (
        <div>
            <p>Welcome</p>
            <ExpenseList/>
        </div>
    );
}

export default ExpenseDashboardPage;