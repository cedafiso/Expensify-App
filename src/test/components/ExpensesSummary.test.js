import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../componentes/ExpensesSummary';

test('Should correctly render Expenses Summary with 1 Expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot();

});

test('Should correctly render Expenses Summary with multiples Expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={20} expensesTotal={2305}/>)
    expect(wrapper).toMatchSnapshot();

});