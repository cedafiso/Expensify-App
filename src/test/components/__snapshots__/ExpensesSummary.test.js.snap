// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render Expenses Summary with 1 Expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expenses totalling 
    $2.35
  </h1>
</div>
`;

exports[`Should correctly render Expenses Summary with multiples Expense 1`] = `
<div>
  <h1>
    Viewing 
    20
     expenses totalling 
    $23.05
  </h1>
</div>
`;
